inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -1180,6 +1180,185 @@ def check_if_uk_in_raw_affiliations(obj, eng):
             return True
 
 
+def check_if_cern_candidate(obj, eng):
+    cern_experiments = [
+        "AMS",
+        "CALICE",
+        "CHIC",
+        "CLEAR",
+        "CLIC",
+        "CLICdp",
+        "CLOUD",
+        "CROWS",
+        "EEE",
+        "EXPLORER",
+        "FASER",
+        "IAXO",
+        "LAGUNA-LBNO",
+        "LARP",
+        "MATHUSLA",
+        "MERIT",
+        "OPAL",
+        "ProtoDUNE-DP",
+        "ProtoDUNE-SP",
+        "SND@LHC",
+        "XSEN",
+    ]
+    cern_collaborations = [
+        "ALICE",
+        "AMS",
+        "ATLAS",
+        "CLEAR",
+        "CLIC",
+        "CLICdp",
+        "CLOUD",
+        "CMS",
+        "COMPASS",
+        "FASER",
+        "FCC",
+        "ISOLDE",
+        "LAGUNA-LBNO",
+        "LHCb",
+        "LHCf",
+        "MATHUSLA",
+        "MEDICIS",
+        "MERIT",
+        "SHINE",
+        "SHiP",
+        "SND@LHC",
+        "TOTEM",
+        "n_TOF",
+    ]
+    non_cern_collaborations = ["CDF", "D0", "NANCY", "nanograv", "PLANCK"]
+    cern_experiment_matches = [
+        {"match": {"metadata.accelerator_experiments.legacy_name": exp}}
+        for exp in cern_experiments
+    ]
+    cern_collaboration_matches = [
+        {"match": {"metadata.collaborations.value": {"query": coll, "operator": "and"}}}
+        for coll in cern_collaborations
+    ]
+    non_cern_collaboration_matches = [
+        {"match": {"metadata.collaborations.value": {"query": coll, "operator": "and"}}}
+        for coll in non_cern_collaborations
+    ]
+
+    should_clauses = (
+        [
+            {"match": {"metadata.corporate_author": "CERN"}},
+            {
+                "match": {
+                    "metadata.authors.affiliations.value": {
+                        "query": "CERN",
+                        "operator": "and",
+                    }
+                }
+            },
+            {
+                "match": {
+                    "metadata.authors.affiliations.value.raw": {
```

Comment:
why we need to check both the raw and the non raw one ?

Reply:
This is how we have it on hep too https://github.com/inspirehep/inspirehep/blob/1c02f3a3fc244aa90a771cf24d08f136c890060e/backend/inspirehep/search/aggregations.py#L332-L460